### DISCUSSION TITLE
Stop Account Creation and Redirect Visitors

### DISCUSSION THREAD

**Discussion by projektug24** [OpnForm/OpnForm#557] [Ideas & Feature Requests]
Hello Community!

First of all, kudos for this amazing project! I have successfully installed it in the Docker variant on my server and am truly impressed by its capabilities. However, I'm facing a small issue: When a visitor fills out a form and accesses the main form URL, such as forms.mydomain.tld, they can create their own account and forms, which I do not want to allow.
I would prefer if this URL simply redirected to the regular visitor homepage or at least disabled the account creation option. Is there already a setting that allows this, or is such a feature planned?

Thank you in advance for your help!

**Comment by JhumanJ** [comment]
Hey there, what version are you using? Can you please make sure that you have `SELF_HOSTED=true` in `api/.env`?

**Reply by projektug24** [reply]
Thank you for your response.
It seems I might have misunderstood a step. I followed the installation instructions as described on [this page](https://docs.opnform.com/deployment/docker). The parameter is set to "true" there. However, I can no longer create an account myself. Either an "Error" appears, or nothing happens.
It also doesn't work when I set the parameter to "false".
However, when I start the Docker container without the .env file using the following command, account creation works at least for everyone:

`docker run -d --name opnform -v $PWD/my-opnform-data:/persist -p 80:80 --restart always jhumanj/opnform`

By the way: Where can I find the version?

**Reply by JhumanJ** [reply]
My bad - we created the docs few days ago and forgot a [section about initial login](https://docs.opnform.com/deployment/docker#initial-login) that I just added back.
What error do you see? Please let us know if this works for you.

**Reply by projektug24** [reply]
Perfect, thank you for the important additional information and the great project. It would be helpful if, when the main page is accessed without a direct form URL, there could be a redirect to my regular homepage. Are there any thoughts or possibilities for implementing this?

**Comment by JhumanJ** [comment]
Just to clarify - if your main domain is `domain.com` and your form domain is `form.domain.com` (e.g. form links would look like `form.domain.com/form/xxx`) you would want `form.domain.com` to redirect to `domain.com`? So basically you'd save the link to the login page? I think this can make sense

**Reply by projektug24** [reply]
Thank you again for the prompt response! :)
Yes, exactly. I would like to use `form.domain.com/login` for logging into the backend. However, if a visitor simply deletes the back part of the URL (e.g., `form.domain.com/form/xxx`), they should be redirected to my main page. This way, I wouldn't have to worry about the imprint, privacy policy, etc., and there would be no confusion due to a website that doesn't match my corporate identity (CI).
One more quick off-topic question: Where exactly do I define hCaptcha as spam protection? I think I need to store the Google Site Key somewhere, but right now I can't see the forest for the trees. ;)

**Reply by JhumanJ** [reply]
Ok makes sense. I added it on our todo. We use hcaptcha, not google recaptcha. You can find details about env here (both env must be changed): https://docs.opnform.com/configuration/environment-variables#configuration-environment-variables

**Reply by projektug24** [reply]
Hi JhumanJ,

I hope you're doing well! I wanted to follow up on the status of the domain redirection feature that you mentioned was added to the roadmap some time ago. Specifically, it was about automatically redirecting visitors from form.domain.com to the main page. As far as I know, this functionality hasn’t been implemented yet.

Are there any updates or a planned timeline for this? If I’ve missed anything, please let me know.

Thank you in advance for your response!

**Reply by projektug24** [reply]
@JhumanJ  
Hi 😊,
I wanted to follow up briefly regarding my last inquiry about the domain redirection feature. Are there any updates on this, or perhaps a timeline for when its implementation can be expected?

**Reply by projektug24** [reply]
Hi @JhumanJ 👋,

Just wanted to quickly follow up about the domain redirection feature – are there any updates on this, or maybe a rough idea when implementation might happen?
Would appreciate any info you have. Thanks a lot!

**Comment by JhumanJ** [comment]
Hey it's available here: https://docs.opnform.com/configuration/subdomain-redirect

**Reply by projektug24** [reply]
It's working perfectly, I hadn't realized the feature was already implemented. Thanks so much for your help!
